include/zephyr/net/wifi_mgmt.h

```diff
@@ -189,17 +177,21 @@ struct wifi_iface_status {
 	unsigned short beacon_interval;
 };
 
+#define WIFI_PS_STATE                  0x01
+#define WIFI_PS_PARAM_MODE             0x02
+#define WIFI_PS_PARAM_TIMEOUT          0x04
+#define WIFI_PS_PARAM_LISTEN_INTERVAL  0x08
+#define WIFI_PS_PARAM_WAKEUP_MODE      0x10
+
 struct wifi_ps_params {
 	enum wifi_ps enabled;
-};
-
-struct wifi_ps_mode_params {
 	enum wifi_ps_mode mode;
-};
-
-struct wifi_ps_timeout_params {
+	char num_twt_flows;
 	int timeout_ms;
-};
+	unsigned short listen_interval;
+	enum wifi_ps_wakeup_mode wakeup_mode;
+	int ps_flag;
```

Comment:
This should be named `oper` or `subcmd`, my earlier review comment to aggregate commands was to simplify the configuration, but this `subcmd` basically re-introduces the sub-commands but using a single structure. Also, the division of parameters doesn't looks correct:

* `num_twt_flows` is in `wifi_ps_params` but `twt_flows` is in `wifi_ps_config`
* `wifi_ps_config` and `wifi_ps_params` are synonymous
* `enabled` and `timeout_ms` in `wifi_ps_params` are dynamic parameters i.e., can be called at any time, but rest of them are only pre-association (mode, listen_interval)

I would suggest below split:

```
/* Static - should be configured before association and cannot be changed during lifetime */
struct wifi_ps_static_config {
   enum wifi_ps_mode mode;
   unsigned short listen_interval;
};

/* Dynamic PS config - Can be send any time, before or after association, and can be changed anytime */
struct wifi_ps_dynamic_config {
  enum wifi_ps enabled;
  unsigned int timeout_ms;
  enum wifi_ps_wakeup_mode wakeup_mode;   
  char num_twt_flows;
  struct wifi_twt_flow_info twt_flows[WIFI_MAX_TWT_FLOWS];
};
```

We could separate out TWT if that makes handling easier.